python/pip_install/pip_repository.bzl

```diff
@@ -61,6 +64,28 @@ def _resolve_python_interpreter(rctx):
             fail("python interpreter `{}` not found in PATH".format(python_interpreter))
     return python_interpreter
 
+def _maybe_set_xcode_location_cflags(rctx, environment):
+    """Patch environment with CPPFLAGS of xcode sdk location.
+
+    Figure out if this interpreter target comes from rules_python, and patch the xcode sdk location if so.
+    Pip won't be able to compile c extensions from sdists with the pre built python distributions from indygreg
+    otherwise. See https://github.com/indygreg/python-build-standalone/issues/103
+    """
+    if (
+        rctx.os.name.lower().startswith("mac os") and
+        rctx.attr.python_interpreter_target != None and
+        rctx.path(Label("@{}//:{}".format(rctx.attr.python_interpreter_target.workspace_name, STANDALONE_INTERPRETER_FILENAME))) and
+        not environment.get(CPPFLAGS)
+    ):
+        xcode_sdk_location = rctx.execute(["xcode-select", "-p"])
+        if xcode_sdk_location.return_code == 0:
+            xcode_root = xcode_sdk_location.stdout.strip()
+            if "commandlinetools" not in xcode_root.lower():
```

Comment:
`commandlinetools` is a magic string here that could use a const to improve the readability here what this `if` block means. Otherwise, a comment would be good to explain the logic here.